### ISSUE TITLE
After how many iterations will I get the result of Yolo-voc.weights?

### ISSUE THREAD

**Issue by zuoqing1988** [AlexeyAB/darknet#253]
Hi, 
I have trained the yolo-voc.cfg for 70,000 iterations, but the IOU is still 52%~53% using the following command

darknet.exe detector recall data/voc.data yolo-voc.cfg backup/yolo-voc_70000.weights

I have also tested the yolo-voc_30000.weights, yolo-voc_40000.weights, yolo_voc_50000.weights, yolo-voc_60000.weights. no better than 53%.

Howerver, the downloaded weights, i.e. yolo-voc.weights, gives IOU:66.18%.

So, after how many iterations will I get the result of yolo-voc.weights?

I compiled the code with VS2015, CUDA9.0, CuDNN7, win10. and run the code on another machine win7.

**Comment by TheMikeyR** [comment]
Are you trying to get yolov1 or yolov2 voc results?

**Comment by zuoqing1988** [comment]
@TheMikeyR 
I tried to reproduce the result of YOLOV2 VOC. But I guess  (but not sure)  I have modified the learning rate in file yolo-voc.cfg.
So, if using the downloaded yolo-voc.cfg, how many iterations can give the result of yolo-voc.weights?
Thank you!

**Comment by TheMikeyR** [comment]
From what I can read in the [paper, page 5-6](https://arxiv.org/pdf/1612.08242.pdf) 

> For VOC we predict 5 boxes with 5 coordinates each and 20 classes per box so 125 filters. We also add a passthrough layer from the final 3 × 3 × 512 layer to the second to last convolutional layer so that our model can use fine grain features. We train the network for **160 epochs** with a starting learning rate of 10−3, dividing it by 10 at 60 and 90 epochs. 
> We use a weight decay of 0.0005 and momentum of 0.9. We use a similar data augmentation to YOLO and SSD with random crops, color shifting, etc. We use the same training strategy on COCO and VOC

One epoch consists of one full training cycle on the training set. Once every sample in the set is seen, you start again - marking the beginning of the 2nd epoch.

voc2007+12 dataset consist of 9963+28952 = 38915 images.
With 160 epochs it means it have to train on 6 226 400 images, if you have a batch size of 64 it will require **97288 iterations**. 
I'm not sure how close you will get to the result though since it is random selecting images for each batch so each training should be a bit differnet, but you should be able to get close.

**Comment by zuoqing1988** [comment]
@TheMikeyR 
I find only 17552 images in train.txt.

details:
train.txt           17552 images    file_size: 1188KB
2007_test.txt     4953 images    file_size:  339KB
2007_train.txt    2502 images    file_size:  171KB
2007_val.txt       2511 images    file_size:  172KB
2012_train.txt    5718 images    file_size:  419KB
2012_val.txt       5824 images    file_size:  427KB

\VOCdevkit\VOC2007\JPEGImages       9963 images
\VOCdevkit\VOC2012\JPEGImages       17125 images

**Comment by TheMikeyR** [comment]
The amount of images is a number i took from VOC
[VOC07](http://host.robots.ox.ac.uk/pascal/VOC/voc2007/)  
>  In total there are **9,963 images** 
[VOC12](http://host.robots.ox.ac.uk/pascal/VOC/voc2012/#data)
> Classification/Detection: The 2012 dataset is the same as that used in 2011. No additional data has been annotated.
[VOC11](http://host.robots.ox.ac.uk/pascal/VOC/voc2011/index.html#data)
> The distributions of images and objects by class are approximately equal across the training/validation and test sets. In total there are **28,952 images**.

I don't know why voc12 are missing like 11k images, I guess you can download the dataset from source and it should be good, else voc have a typo on their page.

**Comment by zuoqing1988** [comment]
Thank you very much! @TheMikeyR

**Comment by TheMikeyR** [comment]
@zuoqing1988 The last images are of course from the test dataset, so my calculations are off. I will update my post with the correct numbers, sorry for the confusion.

**Comment by zuoqing1988** [comment]
@TheMikeyR 
I have found the VOC2012 contains 2007_*.jpg, so the total images are much less than 9963+17125.
And for training, only a part  of VOC2007 are used. So the total images for training are 16552.

The most possible reason for my problem may be I used a learning rate 1e-4 (as in yolo-voc.2.0.cfg).

**Comment by TheMikeyR** [comment]
I guess it is the scaling learning rate you need to get closer.

**Comment by AlexeyAB** [comment]
@zuoqing1988 Hi,

1. Do you use yolo-voc.cfg or yolo-voc.2.0.cfg to train your own model?
2. What `width, height` parameters do you use in your model and in the original model?
3. What `random, learning_rate` did you use for training your model?
4. As you can see, [yolo-voc.weights](https://pjreddie.com/darknet/yolo/) **YOLOv2 544x544 (VOC+COCO)** from the Joseph site trained on both: [Pascal VOC 2007+2012](http://host.robots.ox.ac.uk/pascal/VOC/) dataset and [MS COCO dataset](http://cocodataset.org/#download), therefore, this model is more accurate than your model that trained only on Pascal VOC 2007+2012:

![68747470733a2f2f6873746f2e6f72672f66696c65732f3361362f6664662f6235332f33613666646662353333663334636565396235326264643962623062313964392e6a7067](https://user-images.githubusercontent.com/4096485/32439376-7d096a9a-c2ff-11e7-8194-ef58462562db.jpg)
